src/ray/gcs/redis_context.cc

```diff
@@ -45,74 +46,91 @@ namespace ray {
 
 namespace gcs {
 
-// This is a global redis callback which will be registered for every
-// asynchronous redis call. It dispatches the appropriate callback
-// that was registered with the RedisCallbackManager.
-void GlobalRedisCallback(void *c, void *r, void *privdata) {
-  if (r == nullptr) {
-    return;
+CallbackReply::CallbackReply(redisReply *redis_reply) {
+  RAY_CHECK(nullptr != redis_reply);
+  RAY_CHECK(redis_reply->type != REDIS_REPLY_ERROR) << "Got an error in redis reply: "
```

Comment:
Do we really want the program to crash when the reply is `REDIS_REPLY_ERROR`? It looks like that when the master-slave happens in redis, there will be some error reply. In our internal code, there is a `Status` in the callback argument, the caller needs to handle incorrect status. The users may choose to retry or use `RAY_CHECK` in their callback.

Reply:
One idea is that creating a `Status` object in the constructor. We can form this object when `redis_reply->type != REDIS_REPLY_ERROR` and `REDIS_REPLY_STATUS == redis_reply_->type`, or `Status::OK()` otherwise. The users can call `GetStatus` to check the reply status. I'm not sure whether this is the best way.

Reply:
I agree that every `CallbackReply` contains a `status` member that you suggested above.

Reply:
Sounds good.